File: C1.UWP.FlexChart/VB/StockChart/StockChart/Converters/Boolean2SeriesVisibilityConverter.vb
﻿Public Class Boolean2SeriesVisibilityConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim b As Boolean = System.Convert.ToBoolean(value)
        Return If(b, C1.Chart.SeriesVisibility.Visible, C1.Chart.SeriesVisibility.Hidden)
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Dim v As C1.Chart.SeriesVisibility = DirectCast(value, C1.Chart.SeriesVisibility)
        Return v = C1.Chart.SeriesVisibility.Visible
    End Function
End Class

File: C1.UWP.FlexChart/CS/StockChart/StockChart/Converters/Boolean2SeriesVisibilityConverter.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using Windows.UI.Xaml.Data;

namespace StockChart.Converters
{
    public class Boolean2SeriesVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            bool b = System.Convert.ToBoolean(value);
            return b ? C1.Chart.SeriesVisibility.Visible : C1.Chart.SeriesVisibility.Hidden;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            C1.Chart.SeriesVisibility v = (C1.Chart.SeriesVisibility)value;
            return v == C1.Chart.SeriesVisibility.Visible;
        }
    }
}
